hera_cal/abscal_funcs.py

```diff
@@ -1055,56 +1063,83 @@ def interp2d_vis(model, model_lsts, model_freqs, data_lsts, data_freqs,
 
     data_freqs : 1D array of the data freq axis, dtype=float, shape=(Nfreqs,)
 
-    kind : kind of interpolation method, type=str, options=['linear', 'cubic', ...]
-        see scipy.interpolate.interp2d for details
+    flags : type=DataContainer, dictionary containing model flags. Can also contain model wgts and
+                                will convert appropriately.
+
+    medfilt_flagged : type=bool, if True, run a median filter on flagged model pixels
 
-    fill_value : values to put for interpolation points outside training set
-        if None, values are extrapolated
+    medfilt_window : type=tuple, window for median filter across the (time, freq) axes.
 
-    zero_tol : for amplitudes lower than this tolerance, set real and imag components to zero
+    kx : type=int, type of spline to fit to freq axis of data. See scipy.interpolate.RectBivariateSpline
```

Comment:
Why are we fitting over frequency? Shouldn't the frequencies line up?

Also, is it slower to fit over both time and freq instead of just time (looping over freq)?

Reply:
yeah its a "general purpose" tool assuming you perhaps want to interpolate at a different frequency grid than what's provided. in practice for us it has no effect because the freq grid is the same.

Reply:
Right...but does it slow things down? I don't know whether splines are smart enough to not perform the trival interpolation...

Reply:
i think be definition it is exact. if it fits a line it uses two nearest neighbor points, a quadratic is three and a cubic is four. so my understanding is that by definition its exact.

not sure how much it may be sped up by doing only 1d interpolation. it is already quite quick (relative to other computations) so its not a practical necessity currently...